src/main/java/org/opensearch/neuralsearch/plugin/NeuralSearch.java

```diff
@@ -50,9 +52,14 @@ public class NeuralSearch extends Plugin implements ActionPlugin, SearchPlugin,
      * Gates the functionality of hybrid search
      * Currently query phase searcher added with hybrid search will conflict with concurrent search in core.
      * Once that problem is resolved this feature flag can be removed.
+     * Key is the name string, value is key + transport feature specific prefix, prefix is added by core when we register feature
+     * We need to write and read by the value, key is only for definition
```

Comment:
I think this previous comment having a code link to core was good and would be nice to have it here. 
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/plugins/PluginsService.java#L277